src/ui/src/common/extension_command.ts

```diff
@@ -41,53 +61,52 @@ export declare interface AdapterConvertCommand extends ExtensionCommand {
 }
 
 /** Adapter's "convert" command response. */
-export declare interface AdapterConvertResponse extends ExtensionResponse {
-  graphs?: Graph[];
-  graphCollections?: GraphCollection[];
-}
+export type AdapterConvertResponse = ExtensionResponse;
 
 /** Adapter's "override" command. */
 export declare interface AdapterOverrideCommand extends ExtensionCommand {
   cmdId: 'override';
-  modelPath: string;
   settings: {
     graphs: Graph[];
     changes: ChangesPerNode;
   };
-  deleteAfterConversion: boolean;
 }
 
 /** Adapter's "override" command response. */
-export declare interface AdapterOverrideResponse extends ExtensionResponse {
+export type AdapterOverrideResponse = ExtensionResponse<[{
   success: boolean;
-  graphs?: Graph[];
-}
+}], never>;
 
 /** Adapter's "execute" command. */
 export declare interface AdapterExecuteCommand extends ExtensionCommand {
   cmdId: 'execute';
-  modelPath: string;
-  settings: Record<string, any>;
-  deleteAfterConversion: boolean;
 }
 
-/** Adapter's "execute" command response. */
-export declare interface AdapterExecuteResponse extends ExtensionResponse, ExecutionCommand {
+/** Adapter's "execute" results inside the response. */
+export interface AdapterExecuteResults {
+  stdout: string;
+  log_file: string;
```

Comment:
So my idea was that the logs are going to be sent incrementally as strings with every StatusCheckResponse. Can that work?

Reply:
Yes, we can also add logs to the status check, no problem.